src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,113 @@
+package duke;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.List;
+
+/**
+ * Saves and stores tasks that user has input.
+ */
+public class Storage {
+    private String database;
+    private List<Task> tasks = new ArrayList<>();
+
+    /**
+     * Reads the text file 'database.txt' which contains the saved tasks user input previously.
+     *
+     * @return List of tasks which user had from the last time they used the program.
+     */
+    public List<Task> readFile() throws DukeException {
+        String currentDirectory = System.getProperty("user.dir");
+        File dataFolder = new File(currentDirectory + File.separator + "data");
+        boolean directoryExists = dataFolder.exists() && dataFolder.isDirectory();
+        if (directoryExists) {
+            File database = new File(dataFolder.getAbsolutePath() + File.separator + "database.txt");
+            try {
+                Scanner s = new Scanner(database);
+                while (s.hasNext()) {
+                    String task = s.nextLine();
+                    convertIntoTasks(task);
+                }
+                this.database = database.getAbsolutePath();
+                return tasks;
+            } catch (FileNotFoundException e) {
+                throw new DukeException("No file found sorry please make one");
+            }
+        } else {
+            dataFolder.mkdir();
+            File database = new File(dataFolder.getAbsolutePath() + File.separator + "database.txt");
+            try {
+                database.createNewFile();
+                this.database = database.getAbsolutePath();
+                return tasks;
+            } catch (IOException e) {
+                throw new DukeException("File already exists");
+            }
+        }
+    }
+
+    private void convertIntoTasks(String s) {
```

Comment:
convertIntoTasks() seems to suggest a method that will return a Task.
Maybe change the name to addLoadedData or something similar?

Reply:
Seconded. `addTaskToStorageFromString` might also be a possible option.